### DISCUSSION TITLE
Asks to restart clangd every 5 seconds, even when I say no.

### DISCUSSION THREAD

**Discussion by rnikander** [emacs-lsp/lsp-mode#4021] [Q&A]
Every few seconds I get the following messages from Emacs. It interrupts my editing so it's obnoxious. 

```text
LSP :: clangd has exited (segmentation fault: 11)
Server clangd:84211 exited (check corresponding stderr buffer for details). Do you want to restart it? (y or n) n
```

I say `n` everytime, and then it tries to restart it anyway. Obviously clangd shouldn't be segfaulting, but is there some kind of command to tell LSP mode to stop trying until I tell it to.  I tried `lsp-shutdown-workspace` ... no luck.

**Comment by alexander-naumov** [comment]
While the ideal solution would be to investigate and fix the root cause of the segmentation faults, you can indeed stop lsp-mode from trying to restart the language server automatically. To do this, you can set the lsp-restart option to nil.

You can add the following lines to your Emacs configuration to prevent lsp-mode from automatically trying to restart the language server: `(setq lsp-restart 'ignore)`
